### DISCUSSION TITLE
API rate limit exceeded for public repo with GH_PAT set

### DISCUSSION THREAD

**Discussion by JohannesDeml** [upptime/upptime#305] [Q&A] [answered]
I'm getting a API rate limit exceeded fail for the Uptime CI every few days. Is that to be expected? I set the GH_PAT (tested with and without the new format) in the secrets and most of the time it works... just not always.
Here is the repo with one of the fails: https://github.com/JohannesDeml/upptime/actions/runs/724860941
Any idea why this happens?

**Comment by fraserbooth** [comment]
+1

I've seen this same issue for the first time today.

Has GitHub changed its API Usage Policy?
https://github.blog/2021-04-05-how-we-scaled-github-api-sharded-replicated-rate-limiter-redis/
...suggests that they might have, but then I would expect the issue to effect every call within 60 mins rather than just this Update one.

I assume that the PAT we generate as GH_PAT with "repo" & "workflow" scopes would count as Authenticated, but the error message...
https://github.com/fraserbooth/fb-monitor/actions/runs/735413297
"API rate limit exceeded for 52.251.56.56. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)"
...suggests not...

Is anyone else suffering the same? Does anyone have any ideas what might be causing this?

**Reply by FxllenCode** [reply]
#287...

It is still happening, very annoying. Hopefully this gets resolved.

**Reply by JohannesDeml** [reply]
Still seeing this as well. To give an idea how often this happens, take a look here: https://github.com/JohannesDeml/upptime/actions?query=is%3Afailure++

![grafik](https://user-images.githubusercontent.com/4518980/115347500-00ca5a00-a1a1-11eb-99bf-5cee666c0714.png)

**Comment by AnandChowdhary** [comment, marked as answer]
The fix has been released, but if it's not in your repository, please do this manually:

1. Find and replace `GITHUB_TOKEN` with `GH_PAT` in your `.github/workflows/update-template.yml`
2. Run Update Template CI [manually](https://upptime.js.org/docs/triggers#manual-triggers)
3. This should auto-update your Upptime to the latest version, which both fixes the updating problem and now throws an error in case the update couldn't be pushed.

**Reply by JohannesDeml** [reply]
Thanks for the response, sadly this is not the case in my repo. There is no `GITHUB_TOKEN` in [update-template.yml](https://github.com/JohannesDeml/upptime/blob/master/.github/workflows/update-template.yml) (or anywhere else when searching for it). Also note that I'm getting the error in the Uptime CI, not the Update CI. Might be a different problem?

**Reply by AnandChowdhary** [reply]
We updated some dependencies, so maybe this should be fixed now, maybe it was because of an outdated Octokit? But if it happens again, please let me know!

**Reply by JohannesDeml** [reply]
Awesome, thanks for the quick response! I'll watch out for future failures and will report back in any case.

**Reply by JohannesDeml** [reply]
Just came back to confirm, that I haven't got any messages of broken pipelines anymore, thanks for the fix!
